enhancements/installer/image-based-installer.md

```diff
@@ -0,0 +1,255 @@
+of compatibility. Building a single ISO to boot multiple hosts makes it
+considerably easier for the user to manage. The additional site configuration
+ISO is necessary for configuring each cluster securely and independently.
+
+The user, before running the image-based installer, must generate a [seed image](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/seed-image-generation.md)
+via the [Lifecycle Agent SeedGenerator Custom Resouce (CR)](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/seed-image-generation.md).
+The prerequisites to generating a seed image are the following:
+
+- An already provisioned single-node OpenShift cluster (seed SNO).
+   - The CPU topology of that host must align with the target host(s), i.e. they
+     should have the same number of cores.
+- The [Lifecycle Agent](https://github.com/openshift-kni/lifecycle-agent/tree/main)
+  operator must be installed on the seed SNO.
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+N/A
+
+## Open Questions [optional]
+
+- Should the command-line tool be a subcommand of the OpenShift installer, or a
+  standalone binary?
+
+  Having the functionality provided by the command-line tool in the OpenShift
+  installer would be a natural addition to the latter, as the former refers to
+  the provisioning of single-node OpenShift clusters and generates the
+  required installation artifacts in the same way as the [Agent-based Installer](/enhancements/agent-installer/agent-based-installer.md).
+
+## Test Plan
+
+The image-based installer will be covered by end-to-end testing using virtual
+machines (in a baremetal configuration), automated by some variation on the
+metal platform [dev-scripts](https://github.com/openshift-metal3/dev-scripts/#readme).
+This is similar to the testing of the agent-based installer, the baremetal IPI
+and assisted installation flows.
+
+## Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+N/A
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+N/A
+
+## Support Procedures
+
+N/A
+
+## Alternatives
+
+TBD
```

Comment:
It wont work in a disconnect flow unfortunately

Reply:
Why not? Nothing in the boot ISO building step requires access to the environment where the cluster is running does it? You could download the ISO from outside and carry it in to your disconnected environment.

Reply:
of course. but im not sure what it means from the certification labs perspective. i dont think the iso will be very generic - there will be code to wipe disks, proxy, additional trust bundle, etc.... so it will require us to build new experience in the SaaS.

Reply:
At the time I suggested this I was thinking along the lines of basically each user having one image customised with just their pull secret. But looking at the [other inputs](https://github.com/openshift/installer/pull/8422/files#diff-db815c341f2ef88aeef4c2b74994072c38f9d81d180c9b3b7c7874f0ef9e5cbdR50-R59), there is stuff that is particular to the hardware they are going to install on (the install disk and the Seed image pullspec).

Still, this would only be one form, and a lot more user-friendly than putting the same info into YAML.

One thing that a local tool could potentially do in a disconnected environment that the SaaS can't would be to connect to the registry to verify the seed image. That's the only theoretical advantage I can think of.

Reply:
The same can be said about the Agent based installer option in the openshift installer.
The users execute the openshift-installer binary and get a live-ISO that installs OCP when they boot the nodes with it. 
Yet, the openshift installer is the tool that generates the artifacts to install Openshift (with the exception of the assisted installer discovery image which is basically a pre-provisioning image).
Also, having the `image-based create image` option in the openshift installer can be utilized later by the assisted SaaS if we want to add it there for better UX, though there's no real connection between image-based installation and the assisted install method.